oidc_cli/oidc_impl/client/client.go

```diff
@@ -47,16 +47,20 @@ func NewClient(ctx context.Context, config *Config, clientOptions ...Option) (*C
 		return nil, err
 	}
 
-	oauthConfig := &oauth2.Config{
-		ClientID:    config.ClientID,
-		RedirectURL: fmt.Sprintf("http://localhost:%d", server.GetBoundPort()),
-		Endpoint:    provider.Endpoint(),
-		Scopes: []string{
+	if len(scopes) == 0 {
+		scopes = []string{
 			oidc.ScopeOpenID,
 			oidc.ScopeOfflineAccess,
 			"email",
 			"groups",
-		},
+		}
+	}
```

Comment:
Can you swap this conditional? Set scopes as the default outside the if statement and then if scopes is greater than 0, overwrite it

Reply:
Actually, can you change this implementation? Let's use upper case for the oauthConfig attribute on the Client so we can modify those attributes with a normal Option function.

So you should be able to do something like:

~~~
func ScopeOptions(c *Client) {
c.OIDCConfig.Scopes = []string{"blah", "blah"}
}
~~~

and then pass in the Option to the NewClient as normal, no new scopes argument.